tests/ctst/common/common.ts

```diff
@@ -151,3 +151,89 @@ Then('object {string} should have the user metadata with key {string} and value
         assert(head.Metadata[userMDKey]);
         assert(head.Metadata[userMDKey] === userMDValue);
     });
+
+// add a transition workflow to a bucket
+Given('a transition workflow to {string} location', async function (this: Zenko, location: string) {
+    this.resetCommand();
+    this.addCommandParameter({ bucket: this.getSaved<string>('bucketName') });
+    this.addCommandParameter({
+        lifecycleConfiguration: JSON.stringify({
+            Rules: [
+                {
+                    Status: 'Enabled',
+                    Prefix: '',
+                    Transitions: [
+                        {
+                            Days: 1,
+                            StorageClass: location,
+                        },
+                    ],
+                },
+            ],
+        }),
+    });
+    await S3.putBucketLifecycleConfiguration(this.getCommandParameters());
+});
+
+When('i restore object {string} for {int} days', async function (this: Zenko, objectName: string, days: number) {
+    const objName = objectName ||  this.getSaved<string>('objectName');
+    this.resetCommand();
+    this.addCommandParameter({ bucket: this.getSaved<string>('bucketName') });
+    this.addCommandParameter({ key: objName });
+    const versionId = this.getSaved<Map<string, string>>('createdObjects')?.get(objName);
+    if (versionId) {
+        this.addCommandParameter({ versionId });
+    }
+    this.addCommandParameter({ restoreRequest: `Days=${days}` });
+    await S3.restoreObject(this.getCommandParameters());
+});
+
+// wait for object to transition to a location or get restored from it
+Then('object {string} should be {string} and have the storage class {string}',
+    async function (this: Zenko, objectName: string, operation: string, storageClass: string) {
+        const objName = objectName ||  this.getSaved<string>('objectName');
+        this.resetCommand();
+        this.addCommandParameter({ bucket: this.getSaved<string>('bucketName') });
+        this.addCommandParameter({ key: objName });
+        const versionId = this.getSaved<Map<string, string>>('createdObjects')?.get(objName);
+        if (versionId) {
+            this.addCommandParameter({ versionId });
+        }
+        let conditionOk = false;
+        while (!conditionOk) {
+            const res = await S3.headObject(this.getCommandParameters());
+            if (res.err) {
+                break;
+            }
+            assert(res.stdout);
+            const parsed = safeJsonParse(res.stdout);
+            assert(parsed.ok);
+            const head = parsed.result as {
+                StorageClass: string | undefined,
+                Restore: string | undefined,
+            };
+            const expectedClass = storageClass !== '' ? storageClass : undefined;
+            if (head?.StorageClass === expectedClass) {
+                conditionOk = true;
```

Comment:
* should also verify transition is not in progress
* ideally, we should also verify the intermediate state : e.g. when transitioning/transitioned, we should wait until we indeed reach that "transitioned" state, but also verify that we are in the "transition in progress" state until then (and fail otherwise)

Reply:
This might turn the tests more flaky as we might miss the intermediate state cause it generally is very brief

Reply:
i mean there other way around : if the "final" state is not reached (e.g. conditionOk), then we should be in the matching "transitioning/restoring" state

Reply:
Transitioning state is not visible externally, it's an internal flag in the objecMD.
Added the check for the restore as we can see if an object is currently being restored via a simple `headObject`